### PR TITLE
Add endpoint for downloading Varnam Symbol Table, Removing Word, Cache Improvements

### PR DESCRIPTION
**Pull request by subins2000**
* Use `/languages/identifier/download` to download the language symbol table. Example : https://api.varnamproject.com/languages/ml/download
* `/delete` POST {lang: string, text: string}

### CODE REVIEW

File: http_handlers.go [hunk 1]
```diff
@@ -302,6 +315,8 @@ func handleTrain(c echo.Context) error {
 
 	go func(args trainArgs) { ch <- args }(targs)
 
+	app.cache.Delete(langCode, targs.Pattern)
```
Comment (joicemjoseph) [inline http_handlers.go:318]: why deleting pattern from cache?

File: http_handlers.go [hunk 2]
```diff
@@ -346,6 +361,31 @@ func handleTrainBulk(c echo.Context) error {
...
+
+	c.Request().Header.Set("Content-Type", "application/json")
+
+	if err := c.Bind(&a); err != nil {
+		app.log.Printf("error in binding request details for delete, err: %s", err.Error())
+		return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("error getting metadata. message: %s", err.Error()))
+	}
+
+	_, err := deleteWord(a.LangCode, a.Text)
+	if err != nil {
```
Comment (joicemjoseph) [inline http_handlers.go]: can be replaced by
```go
if _, err := deleteWord(a.LangCode, a.Text); err != nil {
```

Comment (joicemjoseph) [reply http_handlers.go]: add log as well

**Review by joicemjoseph — CHANGES_REQUESTED** [review]
I made some comments. Could you please consider modifying  ?

File: http_handlers.go (hunk 1 above)
Comment (subins2000) [reply http_handlers.go:318]: @joicemjoseph The cache stores the Pattern (english), so it's the english pattern that needs to be removed

File: http_handlers.go (hunk 2 above)
Comment (subins2000) [reply http_handlers.go]: done

**Comment by subins2000** [comment]
This PR is okay now ?

**Comment by joicemjoseph** [comment]
> This PR is okay now ?

Seems ok.
Will check once again. If everything is OK, will merge by today itself.

**Review by joicemjoseph — CHANGES_REQUESTED** [review]
Could you do one last thing ? remove learn from frontend ?


https://github.com/joicemjoseph/varnamd/blob/feat/gomod-echo/ui/javascripts/varnam.js#L77 - 
```&&$.post("learn","{\"text\":\"" + a + "\",\"lang\":\""+B+"\"}")}```

**Comment by subins2000** [comment]
@joicemjoseph If varnamd is run locally without auth, the `/learn` endpoint would still work. So I don't think it should be removed. The current `config.toml` has auth enabled, perhaps we should remove that by default

**Pull request merged**